### PR TITLE
Command Controllers and Command decorators

### PR DESCRIPTION
**Pull request by Brobin**
Ported code from the stale controller branch.

Eliminates the large while loop with nested if statements by creating a Controller class for each page that registers functions with each input command.

In addition, I added a keybinding to the `ENTER` key to open links. This make it easier for users who are navigating using the arrow keys.

### CODE REVIEW

File: rtv/page.py [hunk 1]
```diff
@@ -115,9 +166,19 @@ def __init__(self, stdscr, reddit, content, **kwargs):
         self._content_window = None
         self._subwindows = None
 
+    @BaseController.register('q')
+    def exit(self):
+        sys.exit()
+
+    @BaseController.register('?')
+    def help(self):
+        show_help(self.stdscr)
```
Comment (michael-lazar) [inline rtv/page.py]: Missing import

Comment (Brobin) [inline rtv/page.py]: Fixed with 69a346abe8a1150f4db3eadebeb8a684885002a0

File: rtv/submission.py [hunk 2]
```diff
@@ -91,19 +51,24 @@ def toggle_comment(self):
...
+    def exit_submission(self):
+        self.active = False
 
+    @SubmissionController.register(curses.KEY_F5, 'r')
+    def refresh_content(self):
         url = self.content.name
         self.content = SubmissionContent.from_url(self.reddit, url, self.loader)
         self.nav = Navigator(self.content.get, page_index=-1)
 
+    @SubmissionController.register(curses.KEY_ENTER, 10, 'o')
```
Comment (michael-lazar) [inline rtv/submission.py]: What's the 10 for?

Comment (Brobin) [inline rtv/submission.py]: It's the ascii value of the enter key. The curses.KEY_ENTER wasn't working. I checked the curses documentation and it says KEY_ENTER is unreliable, so I added 10 as a fallback.

Comment (michael-lazar) [inline rtv/submission.py]: ah ok

File: rtv/page.py (hunk 1 above)
Comment (michael-lazar) [inline rtv/page.py]: actually I meant show_help()!

Comment (Brobin) [inline rtv/page.py]: Oh, gotcha.

Comment (Brobin) [inline rtv/page.py]: Fixed 9a4a0b59ea19cf79ee4ae3c5a614211b00a2d3b7

File: rtv/page.py [hunk 3]
```diff
@@ -115,9 +166,19 @@ def __init__(self, stdscr, reddit, content, **kwargs):
...
+
+    @BaseController.register(curses.KEY_UP, 'k')
     def move_cursor_up(self):
         self._move_cursor(-1)
```
Comment (michael-lazar) [inline rtv/page.py]: Looks like these are missing the ``self.clear_input_queue()`` that they used to have.

Basically what that does is prevent holding down the arrow keys or using the scroll wheel from building up a bunch of backlogged commands and lagging the screen once the key is released.

Comment (Brobin) [inline rtv/page.py]: On `move_cursor_up()` and `move_cursor_down()`?

**Comment by michael-lazar** [comment]
Awesome, thanks. I'll let this sit on master for a few days in case anybody discovers any new bugs.

**Comment by Brobin** [comment]
Sounds good. I will also continue to use it / try to break it.

**Pull request merged**